### PR TITLE
Upload the Surefire reports as artifacts

### PR DESCRIPTION
**Pull request by gsmet**
This is part of a pet project I'm working on.

### CODE REVIEW

File: .github/workflows/ci-actions-incremental.yml [hunk 1]
```diff
@@ -40,6 +40,21 @@ env:
   DB_PASSWORD: hibernate_orm_test
   DB_NAME: hibernate_orm_test
 jobs:
+  # this is a hack to work around a GitHub API limitation
```
Comment (famod) [inline .github/workflows/ci-actions-incremental.yml:43]: I'm curious: What's that about?

File: .github/workflows/ci-actions-incremental.yml [hunk 2]
```diff
@@ -715,3 +778,9 @@ jobs:
         with:
           name: test-reports-native-windows-${{matrix.category}}
           path: 'test-reports.tgz'
+      - name: Upload Surefire reports (if build failed)
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: surefire-reports-native-windows-${{matrix.category}}
+          path: "**/target/surefire-reports/TEST-*.xml"
```
Comment (famod) [inline .github/workflows/ci-actions-incremental.yml:786]:
```suggestion
          path: "**/target/*-reports/TEST-*.xml"
```
to also catch failsafe reports. Although not all jobs execute have failsafe tests, it wouldn't hurt to use the broader pattern everywhere.

**Review by famod — COMMENTED** [review]
Maybe also a retetion period of less than the default 90 days would make sense. 🤷‍♂️

File: .github/workflows/ci-actions-incremental.yml (hunk 2 above)
Comment (famod) [reply .github/workflows/ci-actions-incremental.yml:786]: PS: this should work for Gradle:
```
**/build/test-results/test/TEST-*.xml
```

File: .github/workflows/ci-actions-incremental.yml (hunk 1 above)
Comment (gsmet) [reply .github/workflows/ci-actions-incremental.yml]: I added some details in the comment.

File: .github/workflows/ci-actions-incremental.yml (hunk 2 above)
Comment (gsmet) [reply .github/workflows/ci-actions-incremental.yml]: Ok, I generalized a bit but not exactly sure how things will go. We will have to experiment with it.

File: .github/workflows/ci-actions-incremental.yml (hunk 1 above)
Comment (famod) [reply .github/workflows/ci-actions-incremental.yml]: Ok, thanks!

**Comment by gsmet** [comment]
Looks like it worked OK. Lucky to have one test failure for this PR :).

**Comment by gsmet** [comment]
Will merge to see how it goes for the other pull requests.

**Comment by famod** [comment]
Nice! Another gsmetbot. 😁

**Comment by gsmet** [comment]
@famod yeah, this is the first feature for which having the bot in Java was really handy as I was able to reuse some existing library namely https://maven.apache.org/surefire/surefire-report-parser/index.html .

**Comment by famod** [comment]
Really helpful! I think a link to the failed job would make things even more convenient.

**Comment by gsmet** [comment]
Yeah, the thing is that it's not so easy and it would require a lot more experiments as things are a bit decoupled in the GH API.

I'll release the code hopefully next week so people will be able to play with it and add new features if they wish.

**Comment by gastaldi** [comment]
@gsmet I think it would be nice if the Tests Failures were set as the first PR comment, for easier access

**Comment by gastaldi** [comment]
How about displaying the tests summary as a table?

## Tests:    2
| Success      | Failures | Errors| Skipped
| ----------- | ----------- | ----------- | ----------- |
| **0**      | **1**      | 0|**1**|

**Comment by machi1990** [comment]
@gsmet nice feature. 

> How about displaying the tests summary as a table?
> 
> ## Tests: 2
> Success	Failures	Errors	Skipped
> **0**	**1**	0	**1**

+1 @gastaldi  I think it will be a nice improvement.

**Pull request merged**